### ISSUE TITLE
[Bug] bs4 BeautifulSoup - no return value

### ISSUE THREAD

**Issue by JackofSpades707** [xonsh/xonsh#4350]
## xonfig

<details>

```
+------------------+-----------------+
| xonsh            | 0.9.27          |
| Python           | 3.9.5           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 3.0.18          |
| shell type       | prompt_toolkit  |
| pygments         | 2.7.4           |
| on posix         | True            |
| on linux         | True            |
| distro           | arch            |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
| on jupyter       | True            |
| jupyter kernel   | None            |
| xontrib 1        | abbrevs         |
| xontrib 2        | autovox         |
| xontrib 3        | autoxsh         |
| xontrib 4        | avox            |
| xontrib 5        | fzf-widgets     |
| xontrib 6        | jedi            |
| xontrib 7        | prompt_ret_code |
| xontrib 8        | voxapi          |
| xontrib 9        | z               |
+------------------+-----------------+
```

</details>

## Expected Behavior

bs4 should work

## Current Behavior
It errors, no return value for bs4 library use

### Traceback (if applicable)

<details>

```
> ~  $ $XONSH_SHOW_TRACEBACK=True
> ~  $ soup = BeautifulSoup(r.content, 'html.parser')
> ~  $ soup
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 15715, in default
    run_compiled_code(code, self.ctx, None, "single")
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 3147, in run_compiled_code
    func(code, glb, loc)
  File "<xonsh-code>", line 1, in <module>
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 22358, in _pprint_displayhook
    printed_val = pretty(value)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 1652, in pretty
    return obj.xonsh_display()
TypeError: 'NoneType' object is not callable

```

</details>

## Steps to Reproduce

```python
import requests 
from bs4 import BeautifulSoup 
r = requests.get('http://icanhazip.com') 
soup = BeautifulSoup(r.content, 'html.parser') 
print(soup)
```

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by daniel-shimon** [comment]
Hi!
So when xonsh prints an object it checks for a `xonsh_display` function and
calls it if it exists.
Is bs4 mocking this attribute for some reason?

We can add a check to make sure the function is callable and ensure it
returned a string to handle these mocked values.
PRs welcome!

**Comment by dev2718** [comment]
This was resolved in commit 4af6806562f46e1272a1e1ff45fae1bab0a4e5db (issue https://github.com/xonsh/xonsh/issues/4630).
